txscript/data/script_tests.json

```diff
@@ -42,21 +42,516 @@
+["0", "NOTIF ELSE 1 ENDIF", "NONE", "ERR_EMPTY_STACK"],
+["0 0", "NOTIF IF 1 ELSE 0 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0 1", "NOTIF IF 1 ELSE 0 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["1 1", "NOTIF IF 1 ELSE 0 ENDIF ELSE IF 0 ELSE 1 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0 0", "NOTIF IF 1 ELSE 0 ENDIF ELSE IF 0 ELSE 1 ENDIF ENDIF", "NONE", "ERR_EVAL_FALSE"],
+["0 NOTIF", "123", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["NOP", "NOTIF 1 ENDIF", "NONE", "ERR_INVALID_STACK_OPERATION"],
+
+["VERIF test coverage"],
+["0", "IF ELSE 1 ELSE VERIF ENDIF", "NONE", "ERR_RESERVED_OPCODE", "VERIF illegal everywhere"],
+["1","VERIF", "NONE", "ERR_RESERVED_OPCODE", "OP_VERIF is reserved"],
+["0", "IF VERIF ELSE 1 ENDIF", "NONE", "ERR_RESERVED_OPCODE", "VERIF illegal everywhere"],
+
+["VERNOTIF test coverage"],
+["0", "IF ELSE 1 ELSE VERNOTIF ENDIF", "NONE", "ERR_RESERVED_OPCODE", "VERNOTIF illegal everywhere"],
+["1","VERNOTIF", "NONE", "ERR_RESERVED_OPCODE", "OP_VERNOTIF is reserved"],
+["0", "IF VERNOTIF ELSE 1 ENDIF", "NONE", "ERR_RESERVED_OPCODE", "VERNOTIF illegal everywhere"],
 
-["Cat related test coverage"],
+["ENDIF test coverage"],
+["1", "1 ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL", "ENDIF without IF"],
+["1 ELSE 0 ENDIF", "1", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["1", "ELSE ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["1", "ENDIF", "NONE", "ERR_UNBALANCED_CONDITIONAL", "Malformed IF/ELSE/ENDIF sequence"],
+["1", "ENDIF ELSE", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+["1", "ENDIF ELSE IF", "NONE", "ERR_UNBALANCED_CONDITIONAL"],
+
+["VERIFY test coverage"],
+["1 1", "VERIFY", "NONE", "OK"],
+["1", "VERIFY 1", "NONE", "OK"],
+["1 0x05 0x01 0x00 0x00 0x00 0x00", "VERIFY", "NONE", "OK", "values >4 bytes can be cast to boolean"],
+["0", "VERIFY 1", "NONE", "ERR_VERIFY"],
+["1", "VERIFY", "NONE", "ERR_EMPTY_STACK"],
+["1", "VERIFY 0", "NONE", "ERR_EVAL_FALSE"],
+["NOP", "VERIFY 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+
+["RETURN test coverage"],
+["0", "IF RETURN ENDIF 1", "NONE", "OK", "RETURN only works if executed"],
+["1", "IF RETURN ELSE ELSE 1 ENDIF", "NONE", "ERR_EARLY_RETURN", "Multiple ELSEs"],
+["1", "IF 1 ELSE ELSE RETURN ENDIF", "NONE", "ERR_EARLY_RETURN"],
+["1", "RETURN", "NONE", "ERR_EARLY_RETURN"],
+["1", "DUP IF RETURN ENDIF", "NONE", "ERR_EARLY_RETURN"],
+["1", "RETURN 'data'", "NONE", "ERR_EARLY_RETURN", "canonical prunable txout format"],
+["0 IF", "RETURN ENDIF 1", "NONE", "ERR_UNBALANCED_CONDITIONAL", "still prunable because IF/ENDIF can't span scriptSig/scriptPubKey"],
+
+["CHECKLOCKTIMEVERIFY test coverage"],
+["", "CHECKLOCKTIMEVERIFY", "NONE", "ERR_EMPTY_STACK"],
+["1", "CHECKLOCKTIMEVERIFY", "NONE", "OK"],
```

Comment:
Please don't add this test as it can be misleading.  CLTV is already well tested via the transaction tests since it relies on fields in a transaction.

Perhaps replace it with a comment that states such.